.github/workflows/gource-visualization.yml

```
@@ -0,0 +1,94 @@
+              --output-ppm-stream gource_output.ppm \
+              --seconds-per-day 1 \
+              --auto-skip-seconds 0.5 \
+              --highlight-users
+
+            # 4. Convert Gource output to MP4 using FFmpeg
+            ffmpeg -y -r 30 -f image2pipe -vcodec ppm -i gource_output.ppm \
+              -vcodec libx264 -preset medium -crf 18 -threads 0 -bf 0 \
+              gource_output.mp4
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)